### PR TITLE
Activity Log: Integrate new rewind endpoint (remove old one)

### PR DESCRIPTION
**Pull request by elibud**
**Another step on** #7832 

The new `rewind` endpoint provides information about the status of rewind on the site, as well as the last restore operation performed.
In this PR we are only using it for the latter.

**To test:**
1. Go to the Activity Log
2. Do a rewind, check that everything looks as expected.

CC: @dmsnell

### CODE REVIEW

File: WordPressKit/WordPressKit/Activity.swift [hunk 1]
```diff
@@ -157,28 +157,72 @@ public struct ActivityStatus {
...
+    public let id: String
     public let status: Status
-    public let percent: Int
+    public let progress: Int
     public let message: String?
     public let errorCode: String?
     public let failureReason: String?
 
     init(dictionary: [String: AnyObject]) throws {
+        print(dictionary.description)
```
Comment (jklausa) [inline WordPressKit/WordPressKit/Activity.swift]: 😅

File: WordPressKit/WordPressKit/ActivityServiceRemote.swift [hunk 2]
```diff
@@ -52,6 +52,44 @@ public class ActivityServiceRemote: ServiceRemoteWordPressComREST {
...
+    ///
+    /// - Returns: The current rewind status for the site.
+    ///
+    public func getRewindStatus(_ siteID: Int,
+                                success: @escaping (RewindStatus) -> Void,
+                                failure: @escaping (Error) -> Void) {
+        let endpoint = "sites/\(siteID)/rewind"
+        let path = self.path(forEndpoint: endpoint, withVersion: ._2_0)
+        let locale = WordPressComLanguageDatabase().deviceLanguage.slug
+        let parameters: [String: AnyObject] = [
```
Comment (jklausa) [inline WordPressKit/WordPressKit/ActivityServiceRemote.swift]: I think if you declared it as `[String: AnyObject]?` here, you could save the cast when passing it in, but that's a super minor nitpick.

File: WordPressKit/WordPressKit/Activity.swift (hunk 1 above)
Comment (elibud) [reply WordPressKit/WordPressKit/Activity.swift]: 🤦‍♀️

File: WordPressKit/WordPressKit/ActivityServiceRemote.swift (hunk 2 above)
Comment (elibud) [reply WordPressKit/WordPressKit/ActivityServiceRemote.swift]: Thank you for spotting that, fixed now.

**Pull request merged**